packages/core/addon/services/gsheet-export.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2021, Yahoo! Inc.
+ */
+import Service from '@ember/service';
+import { timeout, task } from 'ember-concurrency';
+
+type Options = { pollRateMs?: number; timeoutMs?: number };
+export type GSheetExport = {
+  url: string;
+  fileId: string;
+};
+export type GSheetStatus = {
+  spreadsheetId: string;
+  hasMovedToTeamDrive: boolean;
+  createdTime: string;
+  modifiedTime: string;
+};
+
+export default class GsheetExportService extends Service {
+  POLL_RATE_MS = 5 * 1000;
+  TIMEOUT_MS = 1.8e6;
+
+  /**
+   * Fetches the given URL
+   * @param exportUrl - the url of gsheet export
+   * @param exportPath - the url of the gsheet to be fetched
+   * @param signal - a signal so that the request can be cancelled if need be
+   */
+  async fetchURL(exportUrl: string, exportPath: string, signal: AbortSignal | null = null): Promise<GSheetExport> {
+    const response = await fetch(`${exportUrl}/${exportPath}`, {
```

Comment:
I think we should leave the `/` up to the caller
```suggestion
    const response = await fetch(`${exportUrl}${exportPath}`, {
```